models/models.go

```diff
@@ -643,14 +643,15 @@ func (m *RouteManager) Destroy(guid string) error {
 		return err
 	}
 	var certRow Certificate
-	if err := m.db.Model(route).Related(&certRow, "Certificate").Error; err != nil {
-		return err
-	}
+	if ! m.db.Model(route).Related(&certRow).RecordNotFound() {
+		if err := m.db.Model(route).Related(&certRow, "Certificate").Error; err != nil {
```

Comment:
Why are we keeping this if statement?  If the record is not found, once this always evaluate to true and we'll end up with the same error message regardless?

Reply:
my thought is to catch other errors that we can't handle

Reply:
Ahh, I see, got it.

Reply:
You have a double negative, [!RecordNotFound()](http://gorm.io/docs/error_handling.html#RecordNotFound-Error) means you found it, so in this case you are saying:

If I find the record,
  If I find the related record
    And the error isn't empty
    Return the error
  If I purge the certificate
    And the error is not empty
    Return the error
Return the error

Is that your intended logic? I would recommend not doing `!ResourceNotFound()`, and instead do something like this:

```golang
var certRow Certificate
if errs := db.Model(route).Related(&certRow, "Certificate").GetErrors() {
    // there is at least one error, handle, but now you have []error, so you can do whatever you want.
    return error.New(fmt.Sprintf("%v", errs))
}

// since there are no errors, it means a record was assigned
if err := m.purgeCertificate(route, &certRow); err != nil {
    return err
}
```

Reply:
the double negative is intentional (I think)
if the record is not not found (i.e., it _is_ found), then try to delete it. Otherwise, skip all the stuff about deleting the record.

Reply:
Yeah, the logic had me confused for a bit, but it's sound.